aptos-move/aptos-aggregator/src/aggregator_extension.rs

```diff
@@ -445,6 +474,30 @@ impl AggregatorData {
         }
     }
 
+    pub fn snapshot(&mut self, id: &AggregatorID) -> AggregatorSnapshotID {
+        let snapshot_id = AggregatorSnapshotID {
+            id: self.generate_id(),
+        };
+        let aggregator = self.aggregators.get(id).expect("Aggregator doesn't exist");
```

Comment:
This shouldn't throw but maybe returning result is cleaner

Reply:
Done.